askomics/libaskomics/source_file/SourceFile.py

```diff
@@ -251,36 +275,37 @@ def get_turtle(self, preview_only=False):
 
             # Loop on lines
             for row in tabreader:
-
                 # skip commented lines (# char at the begining)
                 if row[0].startswith('#'):
                     continue
 
-                if len(row) != len(self.get_headers()):
-                    log.error('Invalid line found: %s columns expected, found %s' %s (str(self.get_headers()), str(len(row))))
-                    # TODO catch error somewhere
-                    continue
+                if len(row) != len(self.headers):
+                    e = SourceFileSyntaxError('Invalid line found: %s columns expected, found %s' %s (str(self.headers), str(len(row))))
+                    e.filename = self.path
+                    e.lineno = row_number
+                    log.error(repr(e))
+                    raise e #FIXME: Do we want to read the file anyway ?
 
                 # Create the entity (first column)
                 entity_label = row[0]
                 indent = (len(entity_label) + 1) * " "
-                ttl += ":" + entity_label + " rdf:type :" + self.get_headers[0] + " ;\n"
+                ttl += ":" + entity_label + " rdf:type :" + self.headers[0] + " ;\n"
                 ttl += indent + " rdfs:label \"" + entity_label + "\" ;\n"
 
                 # Add data from other columns
-                for i, header in enumerate(self.get_headers()):
+                for i, header in enumerate(self.headers):
                     current_type = self.forced_column_types[i]
                     #if current_type == 'entity':
                         #relations.setdefault(header, {}).setdefault(entity_label, []).append(row[i]) # FIXME only useful if we want to check for duplicates
                     #else
                     if current_type == 'category':
                         # This is a category, keep track of allowed values for this column
-                        self.category_values.setdefault(header, set()).add(row[i])
+                        self.category_values[header].add(row[i])
 
                     # Create link to value
                     ttl += indent + " :has_" + header + " " + self.delims[current_type][0] + row[i] + self.delims[current_type][1] + " ;\n"
 
-                ttl = attribute_code[:-2] + ".\n"
+                ttl = attribute_code[:-2] + ".\n" #FIXME: Something is missing here
```

Comment:
Ah ok, looking at the old code, it should be "ttl[:-2]"
(not sure if it's still useful though !?)